dataflow/src/main/java/org/checkerframework/dataflow/expression/MethodReferenceTarget.java

```diff
@@ -0,0 +1,87 @@
+package org.checkerframework.dataflow.expression;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.dataflow.qual.Pure;
+
+/**
+ * The right-hand side of a <a
+ * href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.13">Java Method
+ * Reference expression</a>
+ *
+ * <p>The right-hand side of a Java method Reference expression can have the following forms:
+ *
+ * <ul>
+ *   <li>{@literal [TypeArguments] Identifier}, which may be represented by a standard {@link
+ *       JavaExpression}
+ *   <li>{@literal [TypeArguments] "new"}, which is a constructor call and cannot be represented by
+ *       an arbitrary {@link JavaExpression}
+ * </ul>
+ */
+public class MethodReferenceTarget {
+
+  /** The type arguments for this method reference target. */
+  private final @Nullable JavaExpression typeArguments;
+
+  /** The identifier for this method reference target. */
+  private final @Nullable JavaExpression identifier;
+
+  /** Whether this method reference target is a constructor call. */
+  private final boolean isConstructorCall;
+
+  /**
+   * Creates a new method reference target.
+   *
+   * @param typeArguments the type arguments
+   * @param identifier the identifier
+   * @param isConstructorCall whether a method reference target is a constructor call
+   */
+  public MethodReferenceTarget(
+      @Nullable JavaExpression typeArguments,
+      @Nullable JavaExpression identifier,
+      boolean isConstructorCall) {
+    this.typeArguments = typeArguments;
```

Comment:
I suggest checking the validity of the arguments (nullness of the first two vs true of `isConstructorCall`).